Source/WebCore/html/ImageDataArray.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1.  Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ * 2.  Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ * 3.  Neither the name of Apple Inc. ("Apple") nor the names of
+ *     its contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE AND ITS CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL APPLE OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <JavaScriptCore/Uint8ClampedArray.h>
+
+namespace WebCore {
+
+class ImageDataArray {
+public:
+    // FIXME: This API is the minimum subset as currently used by ImageData. More to come...
+    ImageDataArray(Ref<JSC::Uint8ClampedArray>&& data)
+        : m_bufferView(WTFMove(data))
+    { }
+
+    auto length() const
+    {
+        return m_bufferView->length();
+    }
+
+    // FIXME: Remove these temporary functions.
+    Uint8ClampedArray& get() const { return m_bufferView.get(); }
+    auto data() const { return m_bufferView->data(); }
```

Comment:
How is `ImageData` supposed to deal with its method 
```
Uint8ClampedArray& data() const { return m_data.get(); }
```
when the underlying data of this class is `JSC::Float16Array`?

Reply:
As it says just above "// FIXME: Remove these temporary functions." 😉
The goal with this patch was only to introduce this intermediary `ImageDataArray` type that *currently* only deals with `Uint8ClampedArray`. The next patch will add float16 support (WIP: https://github.com/WebKit/WebKit/commit/12134772c758200331c6b268147b91a20afee1b0 ).

Reply:
Basically (spoiler alert!), `ImageDataArray` will store a `variant`, and instead of this `data()` there will be a `visit(F&&)` template function that will call a given lambda with `Uint8ClampedArray` or `Float16Array`.

Reply:
How will the caller know whether `ImageData` stores bytes or float16? I think it will help reviewing these incremental patch if you can put somewhere how `ImageData` will look eventually.

Reply:
Sorry I just opened your WIP: https://github.com/WebKit/WebKit/commit/12134772c758200331c6b268147b91a20afee1b0 now. I saw the answer to my question.